starter-code/TicTacToe/app/src/main/java/ly/generalassemb/drewmahrt/tictactoe/GameActivity.java

```diff
@@ -1,13 +1,176 @@
+                            gameText.setText(p1Name+"'s turn");
+                            p1Turn = !p1Turn;
+                            gameState[tagNum] = 1;
+                        }
+                    }
+                    checkForWinner();
+                    if (gameActive) {
+                        checkIfMovesAvailable();
+                    }
+                    if (!gameActive){
+                        backButton.setVisibility(View.VISIBLE);
+                        backButton.setOnClickListener(new View.OnClickListener() {
+                            @Override
+                            public void onClick(View view) {
+                                gameIntent.putExtra("gamestate", gameState);
+                                finish();
+                            }
+                        });
+                    }
+                }
+            }
+        };
+
+
+        //ArrayList of all imagesView to set background of board.
+        allImageViews.add(imageView0);
+        allImageViews.add(imageView1);
+        allImageViews.add(imageView2);
+        allImageViews.add(imageView3);
+        allImageViews.add(imageView4);
+        allImageViews.add(imageView5);
+        allImageViews.add(imageView6);
+        allImageViews.add(imageView7);
+        allImageViews.add(imageView8);
+
+        //Sets checkered background
+        boolean light = true;
+        for (ImageView view:allImageViews) {
+            if(light){
+                view.setBackgroundResource(R.drawable.wood_bg_light);
+                view.setOnClickListener(selectArea);
+                light=!light;
+            }else{
+                view.setBackgroundResource(R.drawable.wood_bg_dark);
+                view.setOnClickListener(selectArea);
+                light =!light;
+            }
+
+        }
+    }
+
+    public void checkForWinner(){
+        for (int[] winningPosition:winningPositions) {
+            if(gameState[winningPosition[0]]==gameState[winningPosition[1]] //If there's the same value in 3 positions, and that value isn't 2 then someone got 3 in a row.
+                    && gameState[winningPosition[1]]==gameState[winningPosition[2]]
+                    && gameState[winningPosition[0]]!=2){
+                if(gameState[winningPosition[0]]==0){
+                    gameText.setText(p1Name+" is victorious!");
+                    Toast.makeText(this, p1Name+" landed the killing blow!", Toast.LENGTH_SHORT).show();
+                    gameIntent.putExtra("winner",p1Name);
+                    gameIntent.putExtra("loser",p2Name);
+                    setResult(MainActivity.WINNER_CODE,gameIntent);
+                }else{
+
+                    gameText.setText(p2Name+" is victorious!");
+                    Toast.makeText(this, p2Name+ " landed the killing blow!", Toast.LENGTH_LONG).show();
+                    gameIntent.putExtra("winner",p2Name);
+                    gameIntent.putExtra("loser",p1Name);
```

Comment:
I suggest using constants (i.e. `public static final` variables) for your keys rather than hardcode your keys in multiple places.